torch/_inductor/config.py

```diff
@@ -744,6 +744,9 @@ class aot_inductor:
     # rather than embedded into the data section. Needed to support 1B+ parameter models
     force_mmap_weights: bool = False
 
+    # flag to allow buffer mutation. This would remove the read-only property from buffers.
+    allow_buffer_mutation: bool = False
```

Comment:
Can we do like `"0" if is_fbcode() else "1"`? (search for other examples in this file)

Reply:
I don't think this is the right thing, it's not only about fbcode.
If you take a look at the log and previous comment Intel folks made, they moved from .data to .ldata because it overflows to .bss. The overflow is already happening with cases in OSS world. It only happens that the constants they have fits within .ldata so there's no error. If larger constants comes in again, we will get more failures.

Reply:
I agree that diverging the behavior is not a good idea. How about this: instead of adding a config here, Inductor can detect if the model does buffer mutation (https://github.com/pytorch/pytorch/blob/655038687afd19a4a4c9371b77ff046fd6c84be1/torch/_inductor/lowering.py#L5078C25-L5078C41), then decides what section to use and add const_size checking to explicitly fail when size is over limit. We will work on a more comprehensive support in future, but this at least will notice user with a clear error msg.